Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM python:3.9-slim-bullseye
+
+# Set separate working directory for easier debugging.
+WORKDIR /app
+
+# Create virtual environment.
+ENV VIRTUAL_ENV .venv
+RUN python -m venv $VIRTUAL_ENV
+ENV PATH $VIRTUAL_ENV/bin:$PATH
+RUN pip install --upgrade pip setuptools wheel
+
+# Install requirements seperately
+# to take advantage of layer caching.
+COPY requirements.txt .
+RUN pip install --upgrade -r requirements.txt
```

Comment:
Once #1960 is resolved I can replace this line with:

```Dockerfile
COPY setup.cfg .
RUN pip install --upgrade -r <(python -c "import configparser; c = configparser.ConfigParser(); c.read('setup.cfg'); print(c['options']['install_requires'])")
```

Reply:
This will ensure that we install packages from a single source of truth and is consistent with if we were to install the package directly (but with added benefit of layer caching).

Reply:
Implemented this